### PR TITLE
feat: Add contact via chat

### PR DESCRIPTION
**Pull request by SamTremko**

### CODE REVIEW

**Comment by SamTremko** [comment]
Closes #323

File: apps/mobile/src/components/ChatDetailScreen/atoms/index.ts [hunk 1]
```diff
@@ -367,6 +467,247 @@ export const chatMolecule = molecule((getMolecule, getScope) => {
...
+          },
+          () => {
+            // everything OK
+          }
+        )
+      )()
+    }
+  )
+
+  const addRevealedContactWithUiFeedbackAtom = atom(
```
Comment (kaladivo) [inline apps/mobile/src/components/ChatDetailScreen/atoms/index.ts]: Move this atom outside of this molecule and use it when importing via QRcode

File: apps/mobile/src/state/contacts/index.ts [hunk 2]
```diff
@@ -71,6 +71,7 @@ export const importContactFromLinkActionAtom = atom(
     pipe(
       importedContact.numberToDisplay,
       toE164PhoneNumberWithDefaultCountryCode,
+      (a) => a,
```
Comment (kaladivo) [inline apps/mobile/src/state/contacts/index.ts]: !

File: packages/domain/src/general/messaging.ts [hunk 3]
```diff
@@ -35,6 +38,9 @@ export const MessageTypeBackwardCompatible = z.enum([
   'DISAPPROVE_MESSAGING',
   'DELETE_CHAT',
   'BLOCK_CHAT',
+  'REQUEST_CONTACT_REVEAL',
```
Comment (kaladivo) [inline packages/domain/src/general/messaging.ts]: Remove these

**Review by kaladivo — APPROVED** [review]

**Pull request merged**